### PR TITLE
Damage Buff source fix - inventor beehive bug

### PR DESCRIPTION
**Pull request by m888r**
Description:
- Adds an overload for `HandleUnmanaged` that includes the source
- Modifies damage buffs in `BuffComponent.cpp` to also include the source

Motivation and context:
Addresses issue #232 where the surprise beehive inventor skill does not credit kills to the player.

Tested by killing an enemy with the ability and seeing that it now drops loot, which it did not do previously.
[https://streamable.com/btz2vs](https://streamable.com/btz2vs)

### CODE REVIEW

File: dGame/dComponents/SkillComponent.h [hunk 1]
```diff
@@ -161,6 +161,14 @@ class SkillComponent : public Component {
      */
     static void HandleUnmanaged(uint32_t behaviorId, LWOOBJID target);
```
Comment (lcdr) [inline dGame/dComponents/SkillComponent.h:162]: No need to duplicate the entire function, just change this to `static void HandleUnmanaged(uint32_t behaviorId, LWOOBJID target, LWOOBJID originator = LWOOBJID_EMPTY);`

File: dGame/dComponents/SkillComponent.cpp [hunk 2]
```diff
@@ -499,6 +499,25 @@ void SkillComponent::HandleUnmanaged(const uint32_t behaviorId, const LWOOBJID t
 	delete context;
 }
 
+void SkillComponent::HandleUnmanaged(const uint32_t behaviorId, const LWOOBJID target, LWOOBJID source)
+{
+	auto* context = new BehaviorContext(target);
```
Comment (lcdr) [inline dGame/dComponents/SkillComponent.cpp:504]: The constructor already sets the originator, just use `new BehaviorContext(originator)` and don't set the originator afterwards.

**Comment by m888r** [comment]
fixed both please re review when possible

**Comment by lcdr** [comment]
Looks good to me 👍 . I'm not involved with the codebase though, so before merging one of the maintainers (probably @Wincent01) should still check if everything's in accordance with the coding style and if `originator` or `source` is preferred as a name.

**Comment by m888r** [comment]
Thanks, and yeah I couldn't quite figure out which one is correct to use, so that would be very appreciated!

**Comment by averysumner** [comment]
Works on my machine and works on a larger server, merging.

**Pull request merged**